odoo/addons/base/models/res_country.py

```diff
@@ -89,6 +89,11 @@ def _name_search(self, name, domain=None, operator='ilike', limit=None, order=No
         ids = []
         if len(name) == 2:
             ids = list(self._search([('code', 'ilike', name)] + domain, limit=limit, order=order))
+        elif operator in ('=', '!=', 'in', 'not in'):
+            if isinstance(name, str):
+                name = name.capitalize()
+            elif name:  # iterable
+                name = [n.capitalize() for n in name if n is not False]
```

Comment:
Technically, it is possible
1. `name is False`
      But since `len(name) == 2` never cause complaints, we don't have to take care of it in your fix.
3. `False` in `name`
     I am not sure if it will happen in real world. But for stability we should promise the the behavior is not wrose than before in all legal cases
     Maybe `[n.capitalize() if n else n for n in name]` (don't drop `False`)